kbanalysis/src/index.ts

```diff
@@ -1,156 +1,200 @@
 import * as core from "@actions/core"
 import * as github from "@actions/github"
+import { existsSync, fstat, readFileSync } from "fs";
+import { exit } from "process";
 import { createPR } from "./pr_utils";
-import { isKBIssue, getAction, getActionYaml, findToken, printArray, comment, getRunsON, getReadme, checkDependencies, findEndpoints, permsToString, isValidLang, actionSecurity, getTokenInput, normalizePerms} from "./utils"
+import { isKBIssue, getAction, getActionYaml, findToken, printArray, comment, getRunsON, getReadme, checkDependencies, findEndpoints, permsToString, isValidLang, actionSecurity, getTokenInput, normalizePerms, isPaused} from "./utils"
 
 try{
 
     const issue_id = core.getInput("issue-id");
     const token = core.getInput("github-token")
     
     const repos = github.context.repo // context repo
+    const event = github.context.eventName
 
     const client = github.getOctokit(token) // authenticated octokit
-    const resp = await client.rest.issues.get({issue_number: Number(issue_id ), owner: repos.owner, repo:repos.repo})
+    const resp = await client.rest.issues.get({issue_number: Number(issue_id ), owner: repos.owner, repo:repos.repo}) // target issue
 
     const title = resp.data.title // extracting title of the issue.
 
-    if(isKBIssue(title)){
-        core.info("===== Performing analysis =====")
-        const action_name: String = getAction(title) // target action
-        const action_name_split = action_name.split("/") 
-        const target_owner = action_name_split[0]
-        const target_repo = action_name_split.length > 2 ? action_name_split.slice(1,).join("/") : action_name_split[1]
+    if(!isKBIssue(title)){
+        core.info("Not performing analysis as issue is not a valid KB issue")
+        exit(0)
+    }
+
+
+    const action_name: String = getAction(title) // target action
+    const action_name_split = action_name.split("/") 
+    const target_owner = action_name_split[0]
+    const target_repo = action_name_split.length > 2 ? action_name_split.slice(1,).join("/") : action_name_split[1]
 
 
-        const repo_info = await client.rest.repos.get({owner:target_owner, repo: target_repo.split("/")[0]}) // info related to repo.
-        
 
-        let lang:String = ""
-        try{
-            const langs = await client.rest.repos.listLanguages({owner:target_owner, repo:target_repo})
-            lang = Object.keys(langs.data)[0] // top language used in repo
-        }catch(err){
-            lang = "NOT_FOUND"
+    if(resp.data.state === "closed"){
+        if(isPaused(resp.data.labels)){
+            core.info(`Issue creation is paused for ${target_owner}/${target_repo}`)
+            exit(0)
         }
+        const issue_number = 71 // Data Store Issue ID
```

Comment:
is this issue number for this repo or was for your own repo? for this repo, I think the issue is #460.